### ISSUE TITLE
Internal error was raised but Deepsec_api still running

### ISSUE THREAD

**Issue by VincentCheval** [DeepSec-prover/deepsec#51]
Occured on Scytl but the bug is a bit random.

I was using the API and received an internal error from `deepsec_api` but `deepsec_api` kept running (and so the status was still `in progress`). It seems to me that it was an error due to constraint solving, hence that occurred during the solving constraint.

**Comment by VincentCheval** [comment]
```
Please report this error.
Message: [data_structure.ml >> UF.get_unique_unchecked_deduction_fact] There should be only one fact in the pattern to be checked
```

Occurs in fact in the attack simulator.

**Comment by VincentCheval** [comment]
The error does not occur all the time so it's probably due to the computed distribution. Maybe some global variables hasn't been reset correctly.

**Comment by VincentCheval** [comment]
Fix in 58553a6f2132fda17a5696eb495ad071254d4ffb
